applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -374,56 +331,98 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
+        if (NumberOfOpenPipeElements < NumberOfPipeELements) {
+            auto   tip_element = rPipeElements.at(NumberOfOpenPipeElements - 1);
+            double pipe_height = tip_element->GetValue(PIPE_HEIGHT);
 
-            if ((pipe_height > max_pipe_height + std::numeric_limits<double>::epsilon()) ||
+            if ((pipe_height > MaxPipeHeight + std::numeric_limits<double>::epsilon()) ||
                 (pipe_height < pipe_height_accuracy)) {
                 // stable element found; pipe length does not increase during current time step
                 grow = false;
                 tip_element->SetValue(PIPE_EROSION, false);
                 tip_element->SetValue(PIPE_ACTIVE, false);
-                n_open_elements -= 1;
+                NumberOfOpenPipeElements -= 1;
 
                 KRATOS_INFO_IF("PipingLoop", this->GetEchoLevel() > 0 && rank == 0)
-                    << "Number of Open Pipe Elements: " << n_open_elements << std::endl;
+                    << "Number of Open Pipe Elements: " << NumberOfOpenPipeElements << std::endl;
             }
         } else {
             grow = false;
         }
-        return std::make_tuple(grow, n_open_elements);
+        return std::make_tuple(grow, NumberOfOpenPipeElements);
     }
 
     /// <summary>
     /// Saves pipe heights if pipe grows, else reset pipe heights to previous grow step.
     /// </summary>
-    /// <param name="open_pipe_elements"> open pipe elements</param>
-    /// <param name="grow"> boolean to check if pipe grows</param>
+    /// <param name="rOpenPipeElements"> open pipe elements</param>
+    /// <param name="Grow"> boolean to check if pipe grows</param>
     /// <returns></returns>
-    template <typename FilteredElementType>
-    void save_or_reset_pipe_heights(const FilteredElementType& open_pipe_elements, bool grow)
+    template <typename FilteredElementsType>
+    void SaveOrResetPipeHeights(const FilteredElementsType& rOpenPipeElements, bool Grow)
     {
-        for (auto p_element : open_pipe_elements) {
-            if (grow) {
+        for (auto p_element : rOpenPipeElements) {
+            if (Grow) {
                 p_element->SetValue(PREV_PIPE_HEIGHT, p_element->GetValue(PIPE_HEIGHT));
             } else {
                 p_element->SetValue(PIPE_HEIGHT, p_element->GetValue(PREV_PIPE_HEIGHT));
             }
         }
     }
+
+    template <typename FilteredElementsType>
+    void DetermineOpenPipingElements(const FilteredElementsType& rPipingElements, double MaxPipeHeight)
+    {
+        bool       grow                      = true;
+        const auto number_of_piping_elements = rPipingElements.size();
+        auto number_of_open_piping_elements  = this->GetNumberOfActivePipeElements(rPipingElements);
+        while (grow && (number_of_open_piping_elements < number_of_piping_elements)) {
+            // get tip element and activate
+            auto tip_element = rPipingElements.at(number_of_open_piping_elements);
```

Comment:
To comply with the Kratos Style Guide:
```suggestion
            auto p_tip_element = rPipingElements.at(number_of_open_piping_elements);
```

Reply:
done